cilium/CFP-27752-operator-manages-cilium-identities.md

```diff
@@ -0,0 +1,392 @@
+
+Options:
+
+
+
+1. Proposed option - Pod becomes ready immediately after CEP creation and endpoint regeneration. It can still have a temporary ID assigned, while waiting for a global security identity (CID) to be assigned and propagated to other nodes.
+
+    Proposal: To be part of the implementation
+
+
+    Reason: No changes. It can already happen that a pod behind a service is having network policies to incorrectly deny traffic when there is high propagation latency.
+
+
+    The risk of this issue causing disruption is extremely low. Performance of Cilium resource propagation is very high, and only in very rare cases can it happen that it grows more than a few seconds. It’s measured by [CEP propagation delay](https://github.com/cilium/cilium/blob/v1.14/pkg/metrics/metrics.go#L1256), and tracked to be an SLI for network policies. \
+
+
+2. Pod becomes ready only after the CEP with CID assigned is received back at the host. This is only for pods affected by network policies that don’t have an existing matching CID in the CID store. The rest of the pods are ready immediately.
+
+    Proposal: Potential future adjustment \
+Reason: Not related to or required for Operator managing CIDs.
+
+
+    It has existed in DPv2/Cilium for a long time and should be tracked as a separate improvement to network policy enforcement, which is out of the scope of this design document.
+
+
+
+## Two CEP writing sources
+
+When Operator starts managing CIDs, there will be two sources that will be making changes to CEPs, Operator and Agent. This means that Operator and Agent should work in harmony, not racing or overriding each other’s changes.
+
+**Agent**
+
+Create CEP - On pod creation
+
+Update CEP - On pod update. Managing any changes to the CEP, such as Status and labels.
+
+Delete CEP - On pod deletion
+
+Agents will assign CEP’s `Status.Identity` field only when the matching CID is found in the CID watcher cache for new CEPs or when pod labels are changed. On CEP creation, a nil value is assigned, while on CEP updates, the old identity will remain on the CEP until Operator reassigns it.
+
+**Operator**
+
+Update CEP - On CEP or [namespace label changes](#bookmark=kix.kb4ca9sdd34l). Updates only the `Status.Identity`. There will be no collision with Agent updates, because Agent will only update the identity of CEP when it finds a matching CID in the CID watcher cache. In this case, Operator will validate it and not update it again. When pod labels change, Agent will update CEP labels, and only then Operator receive the CEP with changed labels, and update the CID. Operator will also [reconcile all CEPs](#bookmark=id.saz7hxss3be4) every specified interval (e.g. 10 minutes) regardless of any CEP or namespace events.
```

Comment:
Concretely what does "Update the CID" here mean? Which resources are changing? The wording overall in this section is a bit imprecise.

Reply:
You're right. It should be "update the security identity field of the CEP object" instead.

Anyway, we can drop this part from the document, or archive it, because in the "Review discussion" section above, it was proposed for cilium-operator to not update security identity in CEP resource (Proposal: Operator doesn’t update identity field of CEP).

We need to update this in the document.